Export journals using xz instead of gz

samueloph opened the pull request:
As discussed in https://github.com/lepidus/fullJournalTransfer/pull/2, this PR is backwards compatible. This means it'll receive both .gz and .xz files for importation, but will export only .xz files.

Please don't merge yet, this is in WIP state.

samueloph committed:
Export journals using xz instead of gz

As discussed in https://github.com/lepidus/fullJournalTransfer/pull/2, this PR is backwards compatible. This means it'll receive both .gz and .xz files for importation, but will export only .xz files.

ctgraham commented on FullJournalImportExportPlugin.inc.php:
```diff
@@ -308,7 +308,7 @@ function _removeTemporaryFiles($tempfiles) {
 		assert($this->_checkedForTar);
 
-		$tarCommand = Config::getVar('cli', 'tar') . ' -czf ' . escapeshellarg($targetFile);
+		$tarCommand = Config::getVar('cli', 'tar') . ' -cJf ' . escapeshellarg($targetFile);
```
There's no guarantee that a particular tar supports xz compression, whereas gzip compression is (nearly) universal.  See, for example, Solaris tar.  Support should either be detected, or should at the choice of the user.

samueloph replied:
Ok, I'm gonna make it detect support for xz compression.

ctgraham commented on FullJournalImportExportPlugin.inc.php:
```diff
@@ -332,9 +332,7 @@ function tarFiles($targetPath, $targetFile, $sourceFiles) {
-		$tarCommand = Config::getVar('cli', 'tar') . ' -xzf ' . escapeshellarg($targetFile);
-		$tarCommand .= ' -C ' . escapeshellarg($targetPath);
-		exec($tarCommand);
+		exec(Config::getVar('cli', 'tar') . ' -xf ' . escapeshellarg($targetFile) . ' -C ' . escapeshellarg($targetPath));
```
Compression autodetection is a feature of GNU tar.  Other tars will fail here.

samueloph replied:
Guess i'll have to use the switch case again then.
Later today I'm gonna send the fix.

samueloph committed:
Keep OS compatibility

Check for xz support and explicitly uses gz or xz for compression.

samueloph committed:
Identation fix

samueloph committed:
Check xz support at exportation

This one should make the PR usable, but still needs cleanups (like making a function to check for xz support instead of ugly coding).
